Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM golang:1.17-buster AS builder
+
+WORKDIR /build
+
+COPY go.mod go.mod
+COPY go.sum go.sum
+RUN go mod download
+COPY api api
+COPY cmd cmd
+COPY pkg pkg
+COPY main.go main.go
+COPY tools.go tools.go
+COPY Makefile Makefile
+
+RUN make build-cli
+
+FROM golang:1.17-buster
+
+WORKDIR /
+
+COPY --from=builder /build/bin/combo /bin
+
+EXPOSE 8080
+
+CMD ["combo"]
```

Comment:
I was playing around with this second stage locally:

```dockerfile
FROM gcr.io/distroless/static:nonroot
WORKDIR /bin
COPY --from=builder /build/bin/combo .
EXPOSE 8080

CMD ["/bin/combo"]
```

Any thoughts? In order to make it work with distroless final layers, you'd need to do something like the following:

```diff
-RUN make build-cli
-
-FROM golang:1.17-buster
-
-WORKDIR /
-
-COPY --from=builder /build/bin/combo /bin
+RUN CGO_ENABLED=0 go build -o ./bin/combo main.go
 
+FROM gcr.io/distroless/static:nonroot
+WORKDIR /bin
+COPY --from=builder /build/bin/combo .
 EXPOSE 8080
 
-CMD ["combo"]
\ No newline at end of file
+CMD ["/bin/combo"]
```

Reply:
Another thing I just realized: do we need to "force" that combo binary to be rebuilt everytime so we're not caching it's output everytime we build the container? If so, we can either handle that as a `.FORCE: ...` target in the Makefile, or adding the `-a` flag to `go build -a -o ...`. Any thoughts?

Reply:
What would be the point of not caching it? I would think that there would be some logic in Docker/Podman to know when to and not to rebuild the binary. Maybe I'm making a wrong assumption there?

Reply:
We might be fine then if we make the assumption that and change made to the codebase (e.g. api/cmd/pkg/etc directories) would result in a new binary anyways? This is a small detail though so I'm fine with resolving the "force" comment here.

Reply:
We may as well add it just to be explicit about it being new code.